tests/MongoDB.EntityFrameworkCore.FunctionalTests/Utilities/ExtensionMethods.cs

```diff
@@ -36,9 +37,12 @@ public static byte NextByte(this Random random)
     public static short NextInt16(this Random random)
         => (short)random.Next(1, short.MaxValue);
 
-    public static string ToExpectedPrecision(this DateTime dateTime)
-        =>  dateTime.ToString("yyyy-MM-ddTHH:mm:ss.fffK");
-
     public static void WriteTestDocs<T>(this IMongoCollection<T> collection, IEnumerable<T> docs)
         => collection.BulkWrite(docs.Select(p => new InsertOneModel<T>(p)));
+
+    public static DateTime ToBsonPrecision(this DateTime dateTime)
+    {
+        var bsonDateTime = new BsonDateTime(dateTime);
+        return dateTime.Kind == DateTimeKind.Utc ? bsonDateTime.ToUniversalTime() : bsonDateTime.ToLocalTime();
+    }
```

Comment:
Unrelated to `HasBsonType` support. Move to separate PR.

Reply:
This is test-only code and relates to testing DateTime's which I need for the new BsonType tests that touch BsonDateTime. 

I had a ToExpectedPrecision method I was previously using for some other BsonDateTime related tests but it didn't handle UTC. The new method is used for both the new tests and the old.